src/main/java/com/craftingguide/exporter/extensions/forge/ModGatherer.java

```diff
@@ -15,6 +17,11 @@ public void gather(ModPackModel modPack) {
 
         for (ModContainer rawMod : Loader.instance().getActiveModList()) {
             ModModel mod = new ModModel(rawMod);
+            mod.setAuthor(mod.getRawMod().getMetadata().getAuthorList());
+            mod.setDescription(mod.getRawMod().getMetadata().description);
+            mod.setIconicBlock(modPack.getItem(new ItemStack(Blocks.stone)));
+            mod.setUrl(mod.getUrl());
+            mod.setEnabled(true);
```

Comment:
This comes close to satisfying the condition I stated above about needing to properly gather all the data about the mod, but not quite.  Clearly, just using the stone block as the mod's default icon isn't going to work.  Also, if you're setting `enabled` true here, then you don't need the change to the other file.  I'd rather the `enabled` flag still be respected correctly.

Reply:
Also, you'd still need a way to assign sensible groups for the items.  Sometimes mod authors add reasonable groupings by creative tab, but that's almost never sufficient for how Crafting Guide works.

Reply:
For groups, check out how the various other mods work.  Basically, each one has an `Editor` which uses regular expressions to group items together by common themes in their names.  That may not *always* be necessary, but I've yet to find a mod which actually uses creative tabs in a way which produces good results on the website.

Reply:
i made it pick one of the blocks registered from the mod and use that. its atleast better than using stone.